src/dials/algorithms/correlation/plots.py

```diff
@@ -0,0 +1,267 @@
+from __future__ import annotations
+
+import copy
+from collections import OrderedDict
+
+from scipy.cluster import hierarchy
+
+
+def linkage_matrix_to_dict(linkage_matrix):
+    """
+    Convert a linkage matrix to a dictionary
+
+    Args:
+      linkage_matrix (numpy.ndarray): Linkage matrix describing the
+        dendrogram-style linkage from a distance matrix
+    """
+
+    tree = hierarchy.to_tree(linkage_matrix, rd=False)
+
+    d = {}
+
+    # http://w3facility.org/question/scipy-dendrogram-to-json-for-d3-js-tree-visualisation/
+    # https://gist.github.com/mdml/7537455
+
+    def add_node(node):
+        if node.is_leaf():
+            return
+        cluster_id = node.get_id() - len(linkage_matrix) - 1
+        row = linkage_matrix[cluster_id]
+        d[cluster_id + 1] = {
+            "datasets": [i + 1 for i in sorted(node.pre_order())],
+            "height": row[2],
+        }
+
+        # Recursively add the current node's children
+        if node.left:
+            add_node(node.left)
+        if node.right:
+            add_node(node.right)
+
+    add_node(tree)
+
+    return OrderedDict(sorted(d.items()))
+
+
+def scipy_dendrogram_to_plotly_json(ddict):
```

Comment:
I have just realised that this function already exists in `dials/algorithms/clustering/plots.py`, so you can just call that with suitable arguments for the titles.

Reply:
True! Have fixed in commit d1bc2b7